packages/plugin-dom/src/DomParsingEngine.ts

```diff
@@ -1,18 +1,9 @@
-import { ParsingEngine } from '../../plugin-parser/src/ParsingEngine';
-import { DefaultDomParser } from './DefaultDomParser';
+import { XmlParsingEngine } from '../../plugin-xml/src/XmlParsingEngine';
+import { ParsingIdentifier } from '../../plugin-parser/src/ParsingEngine';
 
-export class DomParsingEngine extends ParsingEngine<Node> {
-    static readonly id = 'dom';
-    static readonly defaultParser = DefaultDomParser;
-    /**
-     * Parse a node's attributes and return them.
-     *
-     * @param node
-     */
-    parseAttributes(node: Element): Record<string, string> {
-        return Array.from(node.attributes || []).reduce((attributes, attribute) => {
-            attributes[attribute.name] = attribute.value;
-            return attributes;
-        }, {});
-    }
+export type DomNodeType = HTMLElement | CharacterData;
+
+export class DomParsingEngine<T extends DomNodeType = DomNodeType> extends XmlParsingEngine<T> {
+    static readonly id: ParsingIdentifier = 'dom';
+    static fallbacks = ['xml'];
```

Comment:
I suggest to use the `T1` and `T2` notation we created yesterday. `T1` would be `Node` (for Dom) and `T2` would be `string` (for Xml). This way, the `DomParsingEngine` will be able to parse both Dom nodes and xml string, which I think is the objective. Then I suggest to rename `fallbacks` to `extends` since at this point they are not "fallbacks" per se anymore.